### ISSUE TITLE
Default JSON Schema type validation does not take to account whether a field is required

### ISSUE THREAD

**Issue by Chalolennox** [ngx-formly/ngx-formly#3901]
**Description**
I  am playing around with the json schema examples(https://formly.dev/docs/examples/advanced/json-schema) and realised the schema depedency example with bootstrap does not set validation on the  field **credit_card** correctly when I clear the input after typing in a value. This field is optional. I suspect the validator for number type is active.

**Minimal Reproduction**
```json 
{
    "title": "Schema dependencies",
    "description": "These samples are best viewed without live validation.",
    "type": "object",
    "properties": {
        "simple": {
            "title": "Simple",
            "type": "object",
            "properties": {
                "credit_card": {
                    "type": "number",
                    "title": "Credit card"
                }
            },
            "dependencies": {
                "credit_card": {
                    "properties": {
                        "billing_address": {
                            "type": "string",
                            "title": "Billing address"
                        }
                    },
                    "required": [
                        "billing_address"
                    ]
                }
            }
        }
    }
}
```

I have tried the react version of the library on https://rjsf-team.github.io/react-jsonschema-form/ and it works as intended in that when I clear the **credit_card field**, the form is valid again. As  soon as I type a value and clears it, formly shows a validation error. Here is the active validator:

```javaScript
({
        value
      }) => {
        if (value === void 0) {
          return true;
        }
        if (value === null && types.indexOf("null") !== -1) {
          return true;
        }
        switch (types[0]) {
          case "null": {
            return typeof value === null;
          }
          case "string": {
            return typeof value === "string";
          }
          case "integer": {
            return isInteger(value);
          }
          case "number": {
            return typeof value === "number";
          }
          case "object": {
            return isObject(value);
          }
          case "array": {
            return Array.isArray(value);
          }
        }
        return true;
      }
```

is there a better way to deal with nullable values which are not required when validating?

**Comment by Chalolennox** [comment]
A work around would be to extend **formlyJsonschema.toFieldConfig**

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.3.4](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.3.4) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by Chalolennox** [comment]
Thx, keep up the good work!
